trident/ray_generator.py

```diff
@@ -638,39 +638,31 @@ def _determine_fields_from_ions(ds, ion_list, fields, ftype, sampling_type):
         ion_state = ion[1]
         nuclei_field = "%s_nuclei_mass_density" % atom
         metallicity_field = "%s_metallicity" % atom
-        if ion_state == 1:
```

Comment:
It looks like you forgot to re-alias the N_number_density field to the N_p0_number_density field here.  With this code as it is, it will break backwards compatibility with code that expects there is a N_number_density field on the ray.

Reply:
Sorry, but I'm a little bit confused about what you mean by "aliasing N_number_density field to the N_p0_number_density field". Do you want this block to add both N_number_density field and N_p0_number_density field to the ray (for ion_state ==1)? My only concern about doing that is it would save the same field twice to the ray file and so increase the file size.

Reply:
You removed 6 lines of code which set the alias field and replaced it with a single line of code that does not set the alias field.  I'm  not sure why you removed this.  We still need to be backwards compatible to people who have code expecting to see `X_number_density` and not only `X_p0_number_density`.